### PR TITLE
Feature number of products to buy

### PR DESCRIPTION
**Pull request by ciejj**
As a shopper, I want to choose the number of products I want to buy, so I can add exactly as many items as I need to my shopping cart.

- [x] A shopper can easily change the number of products to purchase before adding them to their cart from the product page
- [x] There is a system & request spec

========
refactor notes
- Cart model has been removed
- Change Cart Add action to CartItem Create action
- Specs have been updated accordingly

### CODE REVIEW

File: app/controllers/cart_items_controller.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      flash[:alert] = result.failure
+    end
+
+    redirect_back(fallback_location: root_path)
+  end
+
+  def destroy_all
+    CartItem.where(user_id: current_user.id).destroy_all
+    flash[:notice] = 'Cart has been emptied'
+    redirect_to root_path
```
Comment (oskarlakner) [inline app/controllers/cart_items_controller.rb]: tutaj mogłbys juz mu przekazac notice po przecinku:
```
redirect_to root_path, notice: 'foo'
```

File: app/services/create_order.rb [hunk 2]
```diff
@@ -38,12 +39,12 @@ def create_order(user)
     end
   end
 
-  def add_products_to_order(order, cart)
-    cart.cart_items.each do |cart_item|
+  def add_products_to_order(order, cart_items)
+    cart_items.each do |cart_item|
       OrderItem.create(product: cart_item.product, order: order)
```
Comment (oskarlakner) [inline app/services/create_order.rb:44]: to chyba chcesz z bangiem wywołać i polegać na mechanizmie transakcji.

File: db/migrate/20210903203639_drop_carts_table.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class DropCartsTable < ActiveRecord::Migration[6.1]
+  def change
```
Comment (oskarlakner) [inline db/migrate/20210903203639_drop_carts_table.rb:2]: @ciejj zastanawiam się jak to Ci umknęło gdy sprawdzałeś ten PR przed wysłaniem. Bo sprawdzałeś, prawda? ;)

**Review by oskarlakner — APPROVED** [review]

File: app/controllers/cart_items_controller.rb (hunk 1 above)
Comment (ciejj) [reply app/controllers/cart_items_controller.rb]: poprawione

File: db/migrate/20210903203639_drop_carts_table.rb (hunk 3 above)
Comment (ciejj) [reply db/migrate/20210903203639_drop_carts_table.rb:2]: to jest bardzo dobre pytanie ;)

**Pull request merged**